nginx-controller/nginx/nginx.go

```diff
@@ -85,13 +93,15 @@ func NewUpstreamWithDefaultServer(name string) Upstream {
 // NewNginxController creates a NGINX controller
 func NewNginxController(nginxConfPath string, local bool) (*NginxController, error) {
 	ngxc := NginxController{
+		nginxEtcPath:   path.Join(nginxConfPath, "etc"),
```

Comment:
I was worried that someone uses `dhparam` as a secret name which would override the certificate secret. (I know this is a edge case, but I do not trust users)
But if we add the namespace to the filename of the certificates we can also use the cert folder for the `dhparam.pem`, this will also solve a not yet raised issue, if a secret with the same name is used in two namespaces.